kml.cc

```diff
@@ -406,34 +406,28 @@ void wpt_icon(xg_string args, const QXmlStreamAttributes*)
 
 void trk_coord(xg_string args, const QXmlStreamAttributes*)
 {
-  int consumed = 0;
-  double lat, lon, alt;
-  Waypoint* trkpt;
-  int n = 0;
-
   route_head* trk_head = route_head_alloc();
-
-  QString iargs = args;
   if (wpt_tmp && !wpt_tmp->shortname.isEmpty()) {
     trk_head->rte_name  = wpt_tmp->shortname;
   }
   track_add_head(trk_head);
-  while ((n = sscanf(CSTRc(iargs), "%lf,%lf,%lf%n", &lon, &lat, &alt, &consumed)) > 0) {
-    trkpt = new Waypoint;
-    trkpt->latitude = lat;
-    trkpt->longitude = lon;
-
-    // Line malformed or two-arg format without alt .  Rescan.
-    if (2 == n) {
-      sscanf(CSTRc(iargs), "%lf,%lf%n", &lon, &lat, &consumed);
-    }
 
-    if (3 == n) {
-      trkpt->altitude = alt;
-    }
+  auto vecs = args.simplified().split(' ');
+  for(const auto& vec : vecs) {
+    QStringList coords = vec.split(',');
+    auto csize = coords.size();
+    auto trkpt = new Waypoint;
 
+    if (csize == 3) {
+      trkpt->altitude = coords[2].toDouble();
+    }
+    if (csize == 2 || csize == 3) {
+      trkpt->latitude = coords[1].toDouble();
+      trkpt->longitude = coords[0].toDouble();
```

Comment:
For read access at() I prefer over operator[] because it will never cause a detach, .i.e.
`coords.at(n)`
etc.
See the discussion at https://doc.qt.io/qt-5/qlist.html#details
However if the list is const then they are equivalent.

Reply:
They're now const.